src/nn/graph_builder.ts

```diff
@@ -317,8 +317,16 @@ export class MLGraphBuilder {
   /**
    * [spec](https://webmachinelearning.github.io/webnn/#dom-mlgraphbuilder-build)
    */
-  build(outputs: MLNamedOperands): MLGraph {
-    return MLGraph.buildAndCompile(outputs);
+  async build(outputs: MLNamedOperands): Promise<MLGraph> {
+    const graph = await MLGraph.buildAndCompile(outputs);
+    return graph;    
+  }
+
+  /**
+   * [spec](https://webmachinelearning.github.io/webnn/#dom-mlgraphbuilder-buildsync)
+   */
+  buildSync(outputs: MLNamedOperands): MLGraph {
```

Comment:
buildSync should be only allowed in dedicated worker, please add a check for that restriction.